### PR TITLE
Convert TimeSequence to use factory map

### PR DESCRIPTION
**Pull request by wthrowe**
Also fixes the doxygen group of the TimeSequences namespace.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/Executables/ExportCoordinates/ExportCoordinates.hpp
Comment (nilsvu) [inline src/Executables/ExportCoordinates/ExportCoordinates.hpp]: Why delete the other triggers? They could potentially be useful to observe coordinates at exactly those times that another simulation observed things at.

Comment (wthrowe) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp]: The triggers don't control the observations, they're just used to terminate the "evolution".  The observations happen after every time step.

The only other trigger that I could see being useful is `SlabCompares`.  I'll add that one back in.

Comment (wthrowe) [reply src/Executables/ExportCoordinates/ExportCoordinates.hpp]: Squashed directly.  Trivial diff:
```patch
diff --git a/src/Executables/ExportCoordinates/ExportCoordinates.hpp b/src/Executables/ExportCoordinates/ExportCoordinates.hpp
index d5c45ab81..241676053 100644
--- a/src/Executables/ExportCoordinates/ExportCoordinates.hpp
+++ b/src/Executables/ExportCoordinates/ExportCoordinates.hpp
@@ -48,6 +48,7 @@
 #include "ParallelAlgorithms/Initialization/Actions/RemoveOptionsAndTerminatePhase.hpp"
 #include "Time/Actions/AdvanceTime.hpp"
 #include "Time/TimeSteppers/TimeStepper.hpp"
+#include "Time/Triggers/SlabCompares.hpp"
 #include "Time/Triggers/TimeCompares.hpp"
 #include "Utilities/Blas.hpp"
 #include "Utilities/ErrorHandling/FloatingPointExceptions.hpp"
@@ -205,7 +206,8 @@ struct Metavariables {
       : tt::ConformsTo<Options::protocols::FactoryCreation> {
     using factory_classes = tmpl::map<
         tmpl::pair<Event, tmpl::list<Events::Completion>>,
-        tmpl::pair<Trigger, tmpl::list<Triggers::TimeCompares>>>;
+        tmpl::pair<Trigger, tmpl::list<Triggers::SlabCompares,
+                                       Triggers::TimeCompares>>>;
   };
 
   enum class Phase { Initialization, RegisterWithObserver, Export, Exit };
```

Answering another possible interpretation of the original question: I had to either remove Triggers from this list or add the TimeSequence factory information, and this change made more sense to me.

**Review by kidder — DISMISSED** [review]

**Comment by wthrowe** [comment]
Rebasing was a bit of a mess.  There's a new commit, and the files in `src/Evolution/Executables/GeneralizedHarmonic/` should be (re)examined.

**Review by kidder — DISMISSED** [review]

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp
Comment (nilsvu) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp:94]: I think this could quickly accumulate levels of nested `add_factory_classes` metafunction calls when adding classes to multiple keys. Would it be reasonable to make this metafunction work a bit like `tmpl::push_back`, taking a parameter pack of key-values to add? It could even be in the `tmpl` namespace. I'm thinking something like this:

```
tmpl::push_back_at<
  original_factory_classes,
  tmpl::pair<Event, tmpl::list<...>>,
  tmpl::pair<Trigger, tmpl::list<...>>>
```

What do you think?

Comment (wthrowe) [reply src/Evolution/Executables/GeneralizedHarmonic/EvolveGeneralizedHarmonicWithHorizon.hpp:94]: I've done the generalization.  I don't think this is generic enough to move to `tmpl`, but we can reevaluate if we find another use for it.

**Review by kidder — COMMENTED** [review]
The fixups LGTM

**Comment by kidder** [comment]
@nilsleiffischer please look at the fixups

**Review by nilsvu — DISMISSED** [review]
LGTM, no need to wait for my re-approval after squashing

**Review by nilsvu — APPROVED** [review]

**Comment by nilsvu** [comment]
Merging now, since all that happened since @kidder's approval was a squash

**Pull request merged**
